examples/tools/scm/git/capture_scm/git_capture_scm.rst

```diff
@@ -0,0 +1,146 @@
+
+        ...
+
+
+We need this code to be in its own Git repository, to see how it works in the real case, so
+please create a folder outside of the ``examples2`` repository, and copy there the contents of the current folder, then:
+
+.. code-block:: text
+
+    $ mkdir /home/myuser/myfolder # or equivalent in other OS
+    $ cp -R . /home/myuser/myfolder # or equivalent in other OS
+    $ cd /home/myuser/myfolder # or equivalent in other OS
+
+    # Initialize the git repo
+    $ git init .
+    $ git add .
+    $ git commit . -m wip
+    # Finally create the package
+    $ conan create .
+    ...
+    hello/0.1: WARN: Current commit 8e8764c40bebabbe3ec57f9a0816a2c8e691f559 doesn't exist in remote origin
+    This revision will not be buildable in other computer
+    hello/0.1: Obtained URL: <local-path>/capture_scm and 8e8764c40bebabbe3ec57f9a0816a2c8e691f559
+    hello/0.1: Copied 1 '.yml' file: conandata.yml
+    hello/0.1: Copied 1 '.py' file: conanfile.py
+    ...
+    hello/0.1: Cloning sources from: {'commit': '8e8764c40bebabbe3ec57f9a0816a2c8e691f559', 'url': '<local-path>/capture_scm'}
+
+Let's explain step by step what is happening:
+
+- When the recipe is exported to the Conan cache, the ``export()`` method executes, running ``scm_url, scm_commit = git.get_url_and_commit()``.
+- This obtains the URL of the repo pointing to the local ``<local-path>/capture_scm`` and the commit ``8e8764c40bebabbe3ec57f9a0816a2c8e691f559``
+- It warns that this information will **not** be enough to re-build from source this recipe once the package is uploaded to the server and is tried to be built from source in other computer, that will not contain the path pointed by ``<local-path>/capture_scm``. This is expected, as the repository that we created doesn't have any remote defined. If our local clone had a remote defined and that remote contained the ``commit`` that we are building, the ``scm_url`` would point to the remote repository instead, making the build from source fully reproducible.
+- The ``export()`` method stores the ``url`` and ``commit`` information in the ``conandata.yml`` for future reproducibility.
+- When the package needs to be built from sources and it calls the ``source()`` method, it recovers the information from the ``conandata.yml`` file, and calls ``git.clone()`` with it to retrive the sources. In this case, it will be cloning from the local checkout in ``<local-path>/capture_scm``, but if it had a remote defined, it will clone from it.
+
+
+.. warning::
+
+    To achieve reproducibility, it is very important for this **scm capture** technique that the current checkout is not dirty. If it was dirty, it would be impossible to guarantee future reproducibility of the build, so ``git.get_url_and_commit()`` can raise errors, and require to commit changes. If more than 1 commit is necessary, it would be recommended to squash those commits before pushing changes to upstream repositories.
```

Comment:
Maybe we can do a second `conan create` to illustrate the fail this or add a `.gitignore` to ignore all the files you don't want to commit?